### PR TITLE
Adding react-window-provider package

### PR DESCRIPTION
**Pull request by dzearing**
This scaffolds the `react-window-provider` package, which provides the following:

Hooks to consume window/document references:

```jsx
const win = useWindow();
const doc = useDocument();
```

And a provider to override defaults:

```jsx
<WindowProvider window={ childWindow } >
   <App />
</WindowProvider>
```

Note this does not integrate it, but that's the idea. We unify on using this, and instruct child window scenarios to use this for pulling window.

### CODE REVIEW

File: packages/react-window-provider/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# @fluentui/react-window-provider
+
+**React Window Provider components for [Fluent UI React](https://developer.microsoft.com/en-us/fluentui)**
+
+These are not production-ready components and **should never be used in product**. This space is useful for testing new components whose APIs might change before final release.
+
+To import React Window Provider components:
+
+```js
+import { ComponentName } from '@fluentui/react-window-provider';
```
Comment (xugao) [inline packages/react-window-provider/README.md:10]: needs some updates here :)

File: packages/react-utilities/README.md [hunk 2]
```diff
@@ -1,17 +1,31 @@
 # @fluentui/react-utilities
 
-**React Utilities for [Fluent UI React](https://developer.microsoft.com/en-us/fluentui) components**
+Utilities for providing and consuming window/document objects.
```
Comment (xugao) [inline packages/react-utilities/README.md:3]: some unexpected changes in `react-utilities` ,  which should go in `window-provider`

File: packages/react-window-provider/README.md (hunk 1 above)
Comment (dzearing) [reply packages/react-window-provider/README.md]: whoops, my changes were not committed! Fixed.

File: packages/react-utilities/README.md (hunk 2 above)
Comment (dzearing) [reply packages/react-utilities/README.md]: Fixed.

**Pull request merged**